### PR TITLE
Fix cmd+down on search result focusing editor

### LINKED ISSUE #68018: CMD+Return/CMD+Down on search result no longer focuses editor
Issue Type: <b>Bug</b>

After Feburary update it's no longer possible to keyboard navigate from a search result into the editor area. This means that doing a Find in Files search can no longer be performed with keyboard only. It's possible to Open Results in Side view but not open in active view.

VS Code version: Code 1.31.0 (7c66f58312b48ed8ca4e387ebd9ffe9605332caa, 2019-02-05T22:32:14.164Z)
OS version: Darwin x64 18.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 x 2200)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (0.58GB free)|
|Process Argv|-psn_0_4092903|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
unique-lines|bib|1.0.0
vscode-eslint|dba|1.8.0
json-pretty-printer|eus|1.1.0
graphql-for-vscode|kum|1.13.0
mssql|ms-|1.4.0
csharp|ms-|1.17.1
Go|ms-|0.9.0
sublime-keybindings|ms-|4.0.0
color-highlight|nau|2.3.0

</details>

### PR DESCRIPTION
**Pull request by roblourens**
Fix #68018

### CODE REVIEW

File: src/vs/platform/list/browser/listService.ts [hunk 1]
```diff
@@ -603,6 +603,17 @@ export interface IResourceResultsNavigationOptions {
 	openOnFocus: boolean;
 }
 
+export interface SelectionKeyboardEvent extends KeyboardEvent {
+	preserveFocus?: boolean;
+}
+
+export function getSelectionKeyboardEvent(typeArg: string, preserveFocus?: boolean): SelectionKeyboardEvent {
+	const e = new KeyboardEvent(typeArg);
```
Comment (roblourens) [inline src/vs/platform/list/browser/listService.ts]: KeyboardEvent does not like to be subclassed

**Review by mjbvz — APPROVED** [review]

**Comment by mjbvz** [comment]
Change looks good to me and seems to work when I built it. @joaomoreno should review as well

**Comment by roblourens** [comment]
@joaomoreno you can merge it if it's good

**Review by joaomoreno — APPROVED** [review]
With this change, we are hard-coding the `list.select` command to never preserve focus... which I'm not entirely sure we should do. Was this true in the latest stable? If so, merge away. Otherwise, I would be careful with it.

---

For the future, it seems to be `list.select` is bogus. We should introduce a `list.open` command which could handle a `preserveFocus` argument.

**Comment by roblourens** [comment]
If I am reading this correctly, that was the case with list.select and the other resource navigator too.

**Comment by Spown** [comment]
this is probably causing #68515

**Pull request merged**